File: python/rpxtokenurl.py
# This code sample shows how to make the auth_info API call using Python.

import requests

# Step 1) Extract the token from your environment.  If you are using app engine,
# you'd do something like:
#
# token = self.request.get('token')
#
# If you are using Django you'd do this:
#
# def rpx_response(request, token):
#   ...
#
# Or, if you are using raw CGI you'd do something like this:
#
# import cgi
# form = cgi.FieldStorage()
# token = form['token'].value

# Step 2) Now that we have the token, we need to make the api call to auth_info.
# auth_info expects an HTTP Post with the following paramters:
engage_api_params = {
    'token': token,
    'apiKey': 'REPLACE_WITH_YOUR_RPX_API_KEY',
    'format': 'json'
}

engage_api_url = 'https://rpxnow.com/api/v2/auth_info'

# make the api call

api_response = requests.get(engage_api_url, params=engage_api_params)

# Step 3) process the json response

auth_info = api_response.json()

# Step 4) use the response to sign the user in

if auth_info['stat'] == 'ok':
    profile = auth_info['profile']
   
    # 'identifier' will always be in the payload
    # this is the unique idenfifier that you use to sign the user
    # in to your site
    identifier = profile['identifier']
   
    # these fields MAY be in the profile, but are not guaranteed. it
    # depends on the provider and their implementation.
    name = profile.get('displayName')
    email = profile.get('email')
    profile_pic_url = profile.get('photo')

    # actually sign the user in.  this implementation depends highly on your
    # platform, and is up to you.
    sign_in_user(identifier, name, email, profile_pic_url)
   
else:
    print 'An error occured: ' + auth_info['err']['msg']

File: php/rpx-token-url.php
<?php
// Below is a very simple and verbose PHP script that implements the Engage
// token URL processing and some popular Pro/Enterprise examples. The code below
// assumes you have the CURL HTTP fetching library with SSL.
require('helpers.php');

ob_start();

// PATH_TO_API_KEY_FILE should contain a path to a plain text file containing
// only your API key. This file should exist in a path that can be read by your
// web server, but not publicly accessible to the Internet.
$janrain_api_key = trim(file_get_contents('PATH_TO_API_KEY_FILE'));

// Set this to true if your application is Pro or Enterprise.
$social_login_pro = false;

// Step 1: Extract token POST parameter
$token = $_POST['token'];

if ($token) {
    // Step 2: Use the token to make the auth_info API call.
    $post_data = array(
        'token' => $token,
        'apiKey' => $janrain_api_key,
        'format' => 'json'
    );

    if ($social_login_pro) {
        $post_data['extended'] = 'true';
    }

    $curl = curl_init();
    $url = 'https://rpxnow.com/api/v2/auth_info';
    $result = curl_helper_post($curl, $url, $post_data);
    if ($result == false) {
        curl_helper_error($curl, $url, $post_data);
        die();
    }
    curl_close($curl);

    // Step 3: Parse the JSON auth_info response
    $auth_info = json_decode($result, true);

    if ($auth_info['stat'] == 'ok') {
        echo "\n auth_info:";
        echo "\n"; var_dump($auth_info);

        // Pro and Enterprise API examples
        if ($social_login_pro) {
            include('social_login_pro_examples.php');
        }

        // Step 4: Your code goes here! Use the identifier in
        // $auth_info['profile']['identifier'] as the unique key to sign the
        // user into your system.

    } else {
        // Handle the auth_info error.
        output('An error occurred', $auth_info);
        output('result', $result);
    }
} else {
    echo 'No authentication token.';
}
$debug_out = ob_get_contents();
ob_end_clean();
?>
<html>
    <head>
        <title>Janrain Token URL Example</title>
    </head>
    <body>
        <pre><?php echo $debug_out; ?></pre>
    </body>
</html>
